src/java/org/apache/cassandra/utils/btree/UpdateFunction.java

```diff
@@ -20,19 +20,36 @@
 
 import java.util.function.BiFunction;
 
-import com.google.common.base.Function;
 /**
- * An interface defining a function to be applied to both the object we are replacing in a BTree and
- * the object that is intended to replace it, returning the object to actually replace it.
+ * An interface defining the method to be applied to the existing and replacing object in a BTree. The objects returned
+ * by the methods will be the object that need to be stored in the BTree.
  */
-public interface UpdateFunction<K, V> extends Function<K, V>
+public interface UpdateFunction<K, V>
 {
     /**
+     * Computes the value that should be inserted in the BTree.
+     *
+     * @param insert the update value
+     * @return the value that should be inserted in the BTree
+     */
+    V insert(K insert);
+
+    /**
+     * Checks if the specified value  should be deleted or not.
+     *
+     * @param existing the existing value to check
+     * @return {@code null} if the value should be removed from the BTree or the existing value if it should not.
+     */
+    V retain(K existing);
```

Comment:
I believe this would also permit us to restore the behaviour of implementing `Function` and save allocations for updateF::insert, but this is a small win, and I don't really mind if we retain the new nomenclature other.